src/v2/codegen/state.rs

```diff
@@ -372,8 +372,12 @@ pub mod util {
 #[cfg(feature = "cli")]
 impl EmitterState {
     /// Sets the crate metadata for this session.
-    pub fn set_meta(&mut self, meta: CrateMeta) {
-        self.crate_meta = Some(Rc::new(RefCell::new(meta)));
+    pub fn set_meta(&self, meta: CrateMeta) {
+        self.crate_meta = Rc::new(RefCell::new(Some(meta)));
+    }
+
+    pub(super) fn get_meta(&self) -> Option<Rc<RefCell<Option<CrateMeta>>>> {
+        Some(self.crate_meta.clone())
```

Comment:
Same about this 🤔

Reply:
Yes, everything involving `crate_meta` field. :smile: